src/A11yDialog.vue

```diff
@@ -73,16 +74,33 @@
       }
     },
 
-    data: () => ({
-      dialog: null
-    }),
+    data () {
+      return {
+        dialog: null
+      }
+    },
 
     methods: {
+      open () {
+        this.$emit('a11y-dialog-opening')
+        this.dialog.show()
+      },
       close () {
+        this.$emit('a11y-dialog-closing')
         this.dialog.hide()
       }
     },
 
+    watch: {
+      'dialog.shown' (val) {
```

Comment:
Where is `dialog.shown` coming from exactly?

Also, since this is Vue 3, we should leverage `import { ref, watch } from 'vue'` idiom.
https://vuejs.org/guide/essentials/watchers.html#basic-example

Another question is how do we deal with when lower-level `a11y-dialog` opens/closes based off the data attributes not instance?